### PR TITLE
`browsingContext.captureScreenshot`: more spec compliance

### PR DESCRIPTION
**Pull request by thiagowfx**
`browsingContext.captureScreenshot`: more spec compliance

- Stop actively focusing the current tab.
- Use Page.getLayoutMetrics to retrieve the screenshot rect clip
- Pass sensible arguments to the CDP method.

Bug: #514

### CODE REVIEW

**Comment by thiagowfx** [comment]
@sadym-chromium This PR fixes a few issues w.r.t. screenshots, however the iframe capture is still not working, even with your previous suggestions :(

File: src/bidiMapper/domains/context/browsingContextImpl.ts [hunk 1]
```diff
@@ -575,13 +575,21 @@ export class BrowsingContextImpl {
   }
 
   async captureScreenshot(): Promise<BrowsingContext.CaptureScreenshotResult> {
-    const [, result] = await Promise.all([
-      // XXX: Either make this a proposal in the BiDi spec, or focus the
-      // original tab right after the screenshot is taken.
-      // The screenshot command gets blocked until we focus the active tab.
-      this.#cdpTarget.cdpClient.sendCommand('Page.bringToFront'),
```
Comment (thiagowfx) [inline src/bidiMapper/domains/context/browsingContextImpl.ts:582]: Upon using layout metrics, this call isn't needed anymore!

Comment (thiagowfx) [reply src/bidiMapper/domains/context/browsingContextImpl.ts:582]: Actually: It's not needed in E2E, but it's still needed in WPT. Odd.

**Review by OrKoN — APPROVED** [review]

**Comment by sadym-chromium** [comment]
> however the iframe capture is still not working, even with your previous suggestions :(

What exactly don't work? Which kind of screenshot is produced? Does it contain the iFrame content? Can the problem be in the framing?

**Comment by thiagowfx** [comment]
> What exactly don't work? Which kind of screenshot is produced? Does it contain the iFrame content? Can the problem be in the framing?

Details here: https://chat.google.com/room/AAAAw7biRSY/SgGyjBGlkBc

TL;DR: Now it seemingly works. Previously it was taking a screenshot of the whole page (i.e. not the iframe only).

**Comment by thiagowfx** [comment]
image

- left: actual screenshot of the iframe (with the current implementation)
- right: reference (the full page)
